File: src/main/java/me/onixdev/util/alert/AlertManager.java
package me.onixdev.util.alert;

import dev.onixac.api.manager.IAlertManager;
import me.onixdev.OnixAnticheat;
import me.onixdev.check.api.Check;
import me.onixdev.user.OnixUser;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.event.HoverEvent;

import java.util.Locale;

public class AlertManager implements IAlertManager {
    private OnixUser user;

    public AlertManager(OnixUser user) {
        this.user = user;
    }


    @Override
    public void toggleAlerts() {
        if (user.isAlertsEnabled()) {
            user.setAlertsEnabled(false);
            user.sendMessage(OnixAnticheat.INSTANCE.getConfigManager().offAlertsMsg.replaceAll("%prefix%",OnixAnticheat.INSTANCE.getConfigManager().getPrefix()));
        }
        else {
            user.setAlertsEnabled(true);
            user.sendMessage(OnixAnticheat.INSTANCE.getConfigManager().onAlertsMsg.replaceAll("%prefix%",OnixAnticheat.INSTANCE.getConfigManager().getPrefix()));
        }
    }


    @Override
    public void toggleVerbose() {
       if (user.isVerboseEnabled()) user.setVerboseEnabled(false);
       else user.setVerboseEnabled(true);
    }

    public void handleVerbose(OnixUser user,Check check, String verbose) {
        String alertString = OnixAnticheat.INSTANCE.getConfigManager().getAlertsformat();
        String prefix = OnixAnticheat.INSTANCE.getConfigManager().getPrefix();
        String hoverMessage = OnixAnticheat.INSTANCE.getConfigManager().getHoverMsg();
        String finalAlertMsg = alertString.replaceAll("%prefix%", prefix)
                .replace("%player%", user.getName())
                .replace("%check_name%", check.getName())
                .replace("%vl%", String.valueOf((int) check.getVl()))
                .replace("%type%", check.getType().toUpperCase(Locale.ROOT))
                .replace("%verbose%", verbose)
                .replace("%experimental%", check.isExperimental() ? " *" : "");
        String finalVerboseMsg = hoverMessage.replaceAll("%player%", user.getName())
                .replace("%check_name%", check.getName().toUpperCase(Locale.ROOT))
                .replace("%type%", check.getType().toUpperCase(Locale.ROOT))
                .replace("%vl%", String.valueOf(check.getVl()))
                .replace("%verbose%", verbose);
        Component alert = Component.text(finalAlertMsg).hoverEvent(HoverEvent.showText(Component.text(finalVerboseMsg)));
        for (OnixUser users: OnixAnticheat.INSTANCE.getPlayerDatamanager().getAllData()) {
            if (users.isVerboseEnabled()) users.sendMessage(alert);
        }

    }

    public void handleAlert(OnixUser user,Check check, String verbose) {
        String alertString = OnixAnticheat.INSTANCE.getConfigManager().getAlertsformat();
        String prefix = OnixAnticheat.INSTANCE.getConfigManager().getPrefix();
        String hoverMessage = OnixAnticheat.INSTANCE.getConfigManager().getHoverMsg();
        String finalAlertMsg = alertString.replaceAll("%prefix%", prefix)
                .replace("%player%", user.getName())
                .replace("%check_name%", check.getName())
                .replace("%type%", check.getType().toUpperCase(Locale.ROOT))
                .replace("%vl%", String.valueOf((int) check.getVl()))
                .replace("%verbose%", verbose)
                .replace("%experimental%", check.isExperimental() ? " *" : "");
        String finalVerboseMsg = hoverMessage.replaceAll("%player%", user.getName())
                .replace("%check_name%", check.getName().toUpperCase(Locale.ROOT))
                .replace("%type%", check.getType().toUpperCase(Locale.ROOT))
                .replace("%vl%", String.valueOf(check.getVl()))
                .replace("%verbose%", verbose);
        Component alert = Component.text(finalAlertMsg).hoverEvent(HoverEvent.showText(Component.text(finalVerboseMsg)));
        for (OnixUser users: OnixAnticheat.INSTANCE.getPlayerDatamanager().getAllData()) {
            if (users.isAlertsEnabled()) users.sendMessage(alert);
        }
    }
}

File: src/main/java/me/onixdev/util/alert/AlertManager.kt
package me.onixdev.util.alert

import dev.onixac.api.manager.IAlertManager
import me.onixdev.OnixAnticheat
import me.onixdev.check.api.Check
import me.onixdev.user.OnixUser
import me.onixdev.util.color.MessageUtil
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.event.HoverEvent
import net.kyori.adventure.text.minimessage.MiniMessage
import org.bukkit.Bukkit

class AlertManager(private val user: OnixUser) : IAlertManager {
    override fun toggleAlerts() {
        if (user.isAlertsEnabled) {
            user.isAlertsEnabled = false
            user.sendMessage(
                OnixAnticheat.INSTANCE.configManager.offAlertsMsg.replace(
                    "%prefix%".toRegex(),
                    OnixAnticheat.INSTANCE.configManager.prefix
                )
            )
        } else {
            user.isAlertsEnabled = true
            user.sendMessage(
                OnixAnticheat.INSTANCE.configManager.onAlertsMsg.replace(
                    "%prefix%".toRegex(),
                    OnixAnticheat.INSTANCE.configManager.prefix
                )
            )
        }
    }


    override fun toggleVerbose() {
        if (user.isVerboseEnabled) user.isVerboseEnabled = false
        else user.isVerboseEnabled = true
    }

    @Suppress("DEPRECATION")
    fun handleVerbose(user: OnixUser, check: Check, verbose: String) {
        val alertString = OnixAnticheat.INSTANCE.configManager.alertsformat
        val prefix = OnixAnticheat.INSTANCE.configManager.prefix
        val finalAlertMsg = alertString.replace("%prefix%".toRegex(), prefix)
            .replace("%player%", user.name)
            .replace("%check_name%", check.name)
            .replace("%vl%", check.getVl().toInt().toString())
            .replace("%type%", check.getType().uppercase())
            .replace("%verbose%", verbose)
            .replace("%experimental%", if (check.isExperimental) " *" else "")
        val alert = MessageUtil.miniMessage(finalAlertMsg).compact()
        if (OnixAnticheat.INSTANCE.configManager.verboseToConsoe) {
            MessageUtil.sendMessage(Bukkit.getConsoleSender(),alert/* MessageUtil.miniMessage(finalAlertMsg).compact()*/)
        }
        if (!OnixAnticheat.INSTANCE.configManager.isFixHoverSystemCompability) {
         //   val alert: Component = MessageUtil.miniMessage(finalAlertMsg)
                //Component.text(finalAlertMsg).hoverEvent(HoverEvent.showText(Component.text(finalVerboseMsg)))
            for (users in OnixAnticheat.INSTANCE.playerDatamanager.allData) {
                if (users.isVerboseEnabled) MessageUtil.sendMessage(users.bukkitPlayer,alert)//users.sendMessage(alert)
            }
        }
        else {
            for (users in OnixAnticheat.INSTANCE.playerDatamanager.allData) {
                if (users.isVerboseEnabled) users.sendMessage(alert)
            }
        }
    }

    fun handleAlert(user: OnixUser, check: Check, verbose: String) {
        val alertString = OnixAnticheat.INSTANCE.configManager.alertsformat
        val prefix = OnixAnticheat.INSTANCE.configManager.prefix
        val hoverMessage = OnixAnticheat.INSTANCE.configManager.hoverMsg
        val finalAlertMsg = alertString.replace("%prefix%".toRegex(), prefix)
            .replace("%player%", user.name)
            .replace("%check_name%", check.name)
            .replace("%vl%", check.getVl().toInt().toString())
            .replace("%type%", check.getType().uppercase())
            .replace("%verbose%", verbose)
            .replace("%experimental%", if (check.isExperimental) "*" else "")
        val alert = MessageUtil.miniMessage(finalAlertMsg).compact()
            MessageUtil.sendMessage(Bukkit.getConsoleSender(),alert /*MessageUtil.miniMessage(finalAlertMsg).compact()*/)
        if (!OnixAnticheat.INSTANCE.configManager.isFixHoverSystemCompability) {
           // val alert: Component = Component.text(finalAlertMsg).hoverEvent(HoverEvent.showText(Component.text(finalVerboseMsg)))
         //   val alert: Component = MessageUtil.miniMessage(finalAlertMsg)
            for (users in OnixAnticheat.INSTANCE.playerDatamanager.allData) {
                if (users.isAlertsEnabled) MessageUtil.sendMessage(users.bukkitPlayer,alert)//users.sendMessage(alert)
            }
        }
        else {
            for (users in OnixAnticheat.INSTANCE.playerDatamanager.allData) {
                if (users.isAlertsEnabled) users.sendMessage(alert)
            }
        }
    }
}